test/common/shouldDo.test.js

```diff
@@ -134,6 +316,115 @@ describe('shouldDo', () => {
     it('returns true if Daily on matching days of the week', () => {
       expect(shouldDo(day, dailyTask, options)).to.equal(true);
     });
+
+    context('Day of the week matches', () => {
+      const weekdayMap = {
+        1: 'm',
+        2: 't',
+        3: 'w',
+        4: 'th',
+        5: 'f',
+        6: 's',
+        7: 'su',
+      };
+      beforeEach(() => {
+        // Set repeat day to current weekday
+        const currentWeekday = moment().isoWeekday();
+        dailyTask.startDate = moment().startOf('day').toDate();
+        dailyTask.repeat = {
+          su: false,
+          s: false,
+          f: false,
+          th: false,
+          w: false,
+          t: false,
+          m: false,
+        };
+        dailyTask.repeat[weekdayMap[currentWeekday]] = true;
+      });
+      context('CDS is midnight (Default dayStart=0)', () => {
+        beforeEach(() => {
+          options.dayStart = 0;
+        });
+        context('Current Date is one day before the matching day', () => {
+          it('should return false', () => {
+            day = moment(day).subtract(1, 'days').toDate();
+            expect(shouldDo(day, dailyTask, options)).to.equal(false);
+          });
+        });
+        context('Current Date is on the matching day', () => {
+          it('returns false if current time is before midnight', () => {
+            day = moment(day).startOf('day').add(1, 'hours').toDate();
```

Comment:
Ah, I think I had a misunderstanding of what CDS is midnight means.

This is what I originally thought is the CDS (today's midnight, at hour 24).
<img src="https://cloud.githubusercontent.com/assets/8214785/25649800/5fb3bdf2-2f8d-11e7-9b0a-43e9029c33f6.jpg" width="350">

When in fact the midnight is actually at hour 0, like this
<img src="https://cloud.githubusercontent.com/assets/8214785/25649819/7e91aee6-2f8d-11e7-9242-e8ba0630657e.jpg" width="350">

So if the task has the start date of today, it should be due today at 1am, which was what my test case was testing. Is this correct?